.github/workflows/build.yml

```diff
@@ -0,0 +1,39 @@
+name: Build image
+on:
+  push:
+    tags:
+      - 'v*'
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build_and_push:
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: read
+    env:
+      REGISTRY: ghcr.io/${{ github.repository_owner }}
+      SOURCE: https://github.com/${{ github.repository }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+      - uses: docker/setup-buildx-action@v3
+      - name: Log in to registry
+        if: github.event_name == 'push'
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build and push tagged image
+        uses: docker/bake-action@v4
+        with:
+          workdir: ./
+          files: docker-compose.yml
+          set: |
+            *.cache-from=type=gha
+            *.cache-to=type=gha,mode=max
+          push: ${{ github.event_name == 'push' && github.ref_type == 'tag' }}
```

Comment:
By taking a look at [Reusing workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows), it seems that we should change all of this to:

```
jobs:
  build_and_push:
    permissions:
      packages: write
      contents: read
    uses: cnpem/epics-in-docker/.github/workflows/ioc-image.yml@vX.Y.Z
    secrets:
      token: ${{ secrets.GITHUB_TOKEN }}
```
And make the `.workflows/ioc-image.yml` in epics-in-docker with version (vX.Y.Z) have the trigger defined as:

```
on:
  workflow_call:
    secrets:
      token:
        required: true
``` 

It is not clear for me if we need or not to explicitly forward the token through `secrets` (as I've described and they show in examples), because they state that:

> When a reusable workflow is triggered by a caller workflow, the github context is always associated with the caller workflow. The called workflow is automatically granted access to github.token and secrets.GITHUB_TOKEN. For more information about the github context, see "[Contexts](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context)."

Reply:
> uses: cnpem/epics-in-docker/.github/workflows/ioc-image.yml@vX.Y.Z

So we could even add that workflow to a prettier path?

I will try to test this in another repo so we don't have to test it in epics-in-docker's `main`.

Reply:
> So we could even add that workflow to a prettier path?

Actually, no. It must be directly in `.github/workflows/` [(no subdirs allowed)][1].

[1]: https://docs.github.com/en/actions/using-workflows/reusing-workflows#creating-a-reusable-workflow

Reply:
I guess we'll find out if the token is required once we try to push an image...